training/modules/hf_gptneox_modules.py

```diff
@@ -14,27 +14,73 @@
 from transformers.models.gpt_neox.modeling_gpt_neox import GPTNeoXLayer as _GPTNeoXBlock
 from transformers.models.gpt_neox.modeling_gpt_neox import GPTNeoXModel as _GPTNeoXModel
 from transformers.models.gpt_neox.configuration_gpt_neox import GPTNeoXConfig as GPTConfig
+from transformers.models.gpt_neox.modeling_gpt_neox import RotaryEmbedding
+
+
+try:
+    from flash_attn.flash_attention import FlashAttention
+    flash_attn_installed = True
+    print('>>>>> flash attention')
+except ImportError:
+    flash_attn_installed = False
+    
+    
 
 def rotate_half(x):
     """Rotates half the hidden dims of the input."""
-    x1 = x[..., : x.shape[-1] // 2]
-    x2 = x[..., x.shape[-1] // 2 :]
+    x1 = x[..., :x.shape[-1] // 2]
+    x2 = x[..., x.shape[-1] // 2:]
```

Comment:
Let's not add reformatting changes to this PR. Leave this out. They make this harder to review.